lib/kafka_producer.js

```diff
@@ -290,41 +336,59 @@ KafkaProducer.prototype.flushEntireCache = function flushEntireCache(callback) {
     }
 };
 
-KafkaProducer.prototype._auditNewMsg = function _auditNewMsg(topic, msgCount) {
+KafkaProducer.prototype._auditNewMsg = function _auditNewMsg(topic, timeBeginInSec, msgCount, topicToMsgcntMaps) {
     if (msgCount > 0) {
-        var self = this;
-        var timeBeginInSec = self._getTimeBeginInSec();
-        if (!self.topicToMsgcntMaps[timeBeginInSec]) {
-            self.topicToMsgcntMaps[timeBeginInSec] = {};
+        if (!topicToMsgcntMaps[timeBeginInSec]) {
```

Comment:
the 'topicToMsgcntMaps' is passed to this method as parameter. Basically, C2 and C3 have their own maps to track count, but use the same methods to calculate those counts and generate JSON auditMsg.